### ISSUE TITLE
Webp image upload only possible with drag and drop

### ISSUE THREAD

**Issue by cceyda** [gradio-app/gradio#4654]
### Describe the bug

when I click to upload an image and the finder opens I can't select images of type webp, they are grayed out.
But when i drag'n'drop the same image it works.
Using chrome browser on mac. (sorry haven't tested with other combos).
Also disabling just based on the filename extension maybe too restrictive(?) there are a lot of webp|gif images on the web that are not actually webp|gif encoded.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

just using a simple image component
`input_image = gr.Image(label="Image", type="pil")`

### Screenshot

<img width="402" alt="image" src="https://github.com/gradio-app/gradio/assets/15624271/4290ba8d-e56b-4c0e-9eb3-d254f75ae667">

### System Info

```shell
Running on huggingface spaces with input images.
Using chrome browser(latest) on mac ventura
```

### Severity

I can work around it

**Comment by pngwn** [comment]
Hmm, this is weird, I thought we were accepting all image mime-types. Will investigate some more.

**Comment by abidlabs** [comment]
Will close for now, we can reopen if this doesn't work for you @cceyda

**Comment by cceyda** [comment]
Yes, I just confirmed it worked! Had noticed it for a while & the space I had tested on was kinda recent so I didn't suspect it (Maybe good to show the gradio version the demo is running as simple as a hovertext on the "Built with Gradio" footer 🤔 ). 
Thanks~
